crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1748,8 +1748,37 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.infer_arguments(arguments);
         self.infer_expression(func);
 
-        // TODO resolve to return type of `func`, if its a callable type
-        Type::Unknown
+        match func.as_ref() {
+            name_expr @ ast::Expr::Name(..) => {
+                let name_expr_id = name_expr.scoped_ast_id(self.db, self.scope);
+                let Some(named_type) = self.types.expressions.get(&name_expr_id) else {
+                    return Type::Unbound;
+                };
+
+                match named_type {
+                    Type::Function(function_type) => {
+                        function_type.returns(self.db).unwrap_or(Type::Unknown)
+                    }
+
+                    // TODO: handle class constructors
+                    // TODO: handle classes which implement the Callable protocol
```

Comment:
This second TODO shouldn't be associated with `Type::Class`, rather it should go with an arm for `Type::Instance`, which is the type representing an instance of a class (as opposed to the class object itself, which `Type::Class` represents.)